File: android-kotlin/QuickStartTasks/app/src/main/java/live/ditto/quickstart/tasks/DittoMode.kt
package live.ditto.quickstart.tasks

/**
 * Identity-mode selection based on env vars.
 *
 * Non-empty `DITTO_OFFLINE_LICENSE_TOKEN` (after trim) selects [OFFLINE];
 * otherwise the app uses [ONLINE_PLAYGROUND].
 */
enum class DittoMode {
    ONLINE_PLAYGROUND,
    OFFLINE;

    companion object {
        fun select(offlineLicenseToken: String?): DittoMode =
            if (offlineLicenseToken.isNullOrBlank()) ONLINE_PLAYGROUND else OFFLINE
    }
}

File: react-native/dittoMode.ts
/**
 * Identity-mode selection based on env vars.
 *
 * Non-empty `DITTO_OFFLINE_LICENSE_TOKEN` (after trim) selects `'offline'`;
 * otherwise the app uses `'online'`.
 */
export type DittoMode = 'online' | 'offline';

export function selectMode(
  offlineLicenseToken: string | null | undefined,
): DittoMode {
  return offlineLicenseToken && offlineLicenseToken.trim().length > 0
    ? 'offline'
    : 'online';
}
